cmake/ci.cmake

```diff
@@ -877,6 +877,22 @@ add_custom_target(ci_cuda_example
     COMMAND ${CMAKE_COMMAND} --build ${PROJECT_BINARY_DIR}/build_cuda_example
 )
 
+###############################################################################
+# Intel C++ Compiler
+###############################################################################
+
+add_custom_target(ci_icpc
+    COMMAND ${CMAKE_COMMAND}
+        -DCMAKE_BUILD_TYPE=Debug -GNinja
+        -DCMAKE_C_COMPILER=icc -DCMAKE_CXX_COMPILER=icpc
+        -DCMAKE_CXX_FLAGS="-DJSON_HAS_FILESYSTEM=0 -DJSON_HAS_EXPERIMENTAL_FILESYSTEM=0"
```

Comment:
Appears as `-DJSON_HAS_FILESYSTEM=0\ -DJSON_HAS_EXPERIMENTAL_FILESYSTEM=0` on the commandline and results in:
```
/__w/json/json/single_include/nlohmann/json.hpp(4116): error: this operator is not allowed in a constant expression
  #elif JSON_HAS_FILESYSTEM
```

Remove? Don't know why `<filesystem>` detection was an issue on ICPC, but might have been fixed by now?